DOCSUP-711: Add description for setting output_format_json_quote_denormals

ka1bi4 opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Detailed description / Documentation draft:

* (ru, en) Added description for setting _output_format_json_quote_denormals_.

ka1bi4 committed:
DOCSUP-711: (en) In settings added description for output_format_json_quote_denormals.

ka1bi4 committed:
DOCSUP-711: Fix invalid anchor.

ka1bi4 committed:
DOCSUP-711: (ru) In settings added description for output_format_json_quote_denormals.

ka1bi4 committed:
DOCSUP-711: Change values styles.

BayoNet commented:
1. Add a link from the setting description to the JSON format description.
2. Add a link from the JSON format description to the setting description.

Add examples of output with the setting turned off and on.

ka1bi4 committed:
DOCSUP-711: Add a link from the setting description to the JSON format description. (by pr comment)
Add a link from the JSON format description to the setting description. (by pr comment)

ka1bi4 committed:
DOCSUP-711: Rewrite sentence.

ka1bi4 committed:
DOCSUP-711: (ru) Added link from settings.md to JSON format.

ka1bi4 committed:
DOCSUP-711: (en) Added example of output with the setting (by pr comment).

ka1bi4 committed:
DOCSUP-711: Add example of output with the setting.

ka1bi4 committed:
DOCSUP-711: Minor fix.

ka1bi4 committed:
DOCSUP-711: Update example of settings using.

ka1bi4 committed:
DOCSUP-711: Minor fix.

BayoNet commented on docs/en/interfaces/formats.md:
```diff
@@ -464,7 +464,7 @@ If the query contains GROUP BY, rows\_before\_limit\_at\_least is the exact numb
 This format is only appropriate for outputting a query result, but not for parsing (retrieving data to insert in a table).
 
-ClickHouse supports [NULL](../sql-reference/syntax.md), which is displayed as `null` in the JSON output.
+ClickHouse supports [NULL](../sql-reference/syntax.md), which is displayed as `null` in the JSON output. Set the [output\_format\_json\_quote\_denormals](../operations/settings/settings.md#settings-output_format_json_quote_denormals) to 1, to enables `+nan`, `-nan`, `+inf`, `-inf` values in output.
```
```suggestion
ClickHouse supports [NULL](../sql-reference/syntax.md), which is displayed as `null` in the JSON output. To enable `+nan`, `-nan`, `+inf`, `-inf` values in output, set the [output\_format\_json\_quote\_denormals](../operations/settings/settings.md#settings-output_format_json_quote_denormals) to 1.
```

BayoNet commented on docs/en/operations/settings/settings.md:
```diff
@@ -998,6 +998,99 @@ The results of the compilation are saved in the build directory in the form of .
+
+Default value: 0.
+
+For example, when this setting disabled, the query returns null values in output:
```
```suggestion
**Example**

Consider the following table:

```
Here is the contents of the test_table.
Maybe we can find a more interesing name for the table?
```

When `output_format_json_quote_denormals = 0`, the query returns null values in output:
```

ka1bi4 committed:
Update docs/en/interfaces/formats.md by pr

Co-authored-by: BayoNet <da-daos@yandex.ru>

ka1bi4 committed:
[wip]Update docs/en/operations/settings/settings.md

Co-authored-by: BayoNet <da-daos@yandex.ru>

ka1bi4 committed:
DOCSUP-711: (en) Add sample table and fix pr comments.

ka1bi4 committed:
DOCSUP-711: (ru) Updated from en and by pr comments.

ka1bi4 committed:
DOCSUP-711: (ru) Updated from en and by pr comments.

ka1bi4 committed:
DOCSUP-711: Added name for sample table.

BayoNet commented on docs/en/operations/settings/settings.md:
```diff
@@ -998,6 +998,105 @@ The results of the compilation are saved in the build directory in the form of .
+                {
+                        "divide(area, period)": null
+
+        ],
```
Missing "}" ?

BayoNet commented on docs/en/operations/settings/settings.md:
```diff
@@ -998,6 +998,105 @@ The results of the compilation are saved in the build directory in the form of .
+                },
+                {
+                        "divide(area, period)": "-inf"
+
```
Missing "}" ?

BayoNet commented on docs/ru/interfaces/formats.md:
```diff
@@ -432,7 +432,7 @@ JSON совместим с JavaScript. Для этого, дополнитель
 Этот формат подходит только для вывода результата выполнения запроса, но не для парсинга (приёма данных для вставки в таблицу).
 
-ClickHouse поддерживает [NULL](../sql-reference/syntax.md), который при выводе JSON будет отображен как `null`.
+ClickHouse поддерживает [NULL](../sql-reference/syntax.md), который при выводе JSON будет отображен как `null`. Чтобы включить значения `+nan`, `-nan`, `+inf`, `-inf`, установите параметр [output\_format\_json\_quote\_denormals](../operations/settings/settings.md#settings-output_format_json_quote_denormals) равным 1.
```
```suggestion
ClickHouse поддерживает [NULL](../sql-reference/syntax.md), который при выводе JSON будет отображен как `null`. Чтобы включить отображение в результате значений  `+nan`, `-nan`, `+inf`, `-inf`, установите параметр [output\_format\_json\_quote\_denormals](../operations/settings/settings.md#settings-output_format_json_quote_denormals) равным 1.
```

BayoNet commented on docs/ru/operations/settings/settings.md:
```diff
@@ -908,6 +908,105 @@ load_balancing = first_or_random
 
+## output\_format\_json\_quote\_denormals {#settings-output_format_json_quote_denormals}
+
+При включении, позволяет выводить значения `+nan`, `-nan`, `+inf`, `-inf` в формате выходных данных [JSON](../../interfaces/formats.md#json).
```
```suggestion
При выводе данных в формате [JSON](../../interfaces/formats.md#json) включает отображение значений `+nan`, `-nan`, `+inf`, `-inf`.
```

ka1bi4 committed:
Update docs/ru/operations/settings/settings.md by pr comment

Co-authored-by: BayoNet <da-daos@yandex.ru>

ka1bi4 committed:
Update docs/ru/interfaces/formats.md  by pr comment

Co-authored-by: BayoNet <da-daos@yandex.ru>

ka1bi4 committed:
Update settings.md

ka1bi4 committed:
Update settings.md
